packages/runtime/src/runtime/menu.tsx

```diff
@@ -160,8 +268,112 @@ export const SideBarMenu: React.FC<MenuBaseProps> = ({ id, ...props }) => {
     }
   }, [location.pathname, props.items]);
 
+  return (
+    <Col
+      id="ensemble-sidebar"
+      style={{
+        backgroundColor,
+        borderRight: "1px solid lightgrey",
+        display: "flex",
+        justifyContent: "space-between",
+        flexDirection: "column",
+        width: isCollapsed ? undefined : props.styles?.width,
+        height: "100vh",
+        alignItems: "center",
+        zIndex: 1,
+      }}
+    >
+      {props.header ? EnsembleRuntime.render([props.header]) : null}
+      {ItemsMenu(
+        values?.items || [],
+        values?.styles || {},
+        selectedItem,
+        setSelectedItem,
+        values?.isCollapsed,
+      )}
+      {props.footer ? EnsembleRuntime.render([props.footer]) : null}
+    </Col>
+  );
+};
+
+export const DrawerMenu: React.FC<DrawerMenuProps> = ({
+  id,
+  items,
+  header,
+  footer,
+  styles,
+  isOpen,
+  setOpen,
+  onCollapse,
+}) => {
+  const validPosition = ["left", "right", "top", "bottom"];
+
+  const [selectedItem, setSelectedItem] = useState<string | undefined>();
+
+  const { rootRef } = useRegisterBindings(
+    {
+      open: isOpen,
+      height: styles?.height,
+      width: styles?.width,
+      position: styles?.position,
+    },
+    id,
+    {
+      openDrawer: () => setOpen(true),
+      closeDrawer: () => setOpen(false),
+    },
+    {
+      debounceMs: 300,
+    },
+  );
+
+  const handleClose = (): void => {
+    setOpen(false);
+    onCollapseCallback();
+  };
+
+  const onCollapseAction = useEnsembleAction(onCollapse);
+  const onCollapseCallback = useCallback(() => {
+    if (!onCollapseAction) return;
+    return onCollapseAction.callback();
+  }, [onCollapseAction]);
```

Comment:
There's a lot of shared logic/hooks here that should be hoisted into the parent menu component, and the Drawer and Sidebar should just be responsible for rendering the views. So basically take all the logic in sidebar menu and put it into the parent menu component.